zerver/lib/export.py

```diff
@@ -2196,7 +2216,7 @@ def export_files_from_s3(
         email_gateway_bot = get_system_bot(settings.EMAIL_GATEWAY_BOT, internal_realm.id)
         user_id_to_email[email_gateway_bot.id] = email_gateway_bot.email
 
-    def iterate_attachments() -> Iterator[Record]:
+    def iterate_attachments(download_obj: Callable[[str], Any]) -> Iterator[Record]:
```

Comment:
Maybe `do_download_object` would be a better name for this? The current name sounds a bit like it's a "download object", not a function that downloads the object.